### DISCUSSION TITLE
Second Order SCF Stability

### DISCUSSION THREAD

**Discussion by zh-yichi** [pyscf/pyscf#3258] [Q&A]
Hi, I found that some times SOSCF results are not internally stable. Like this one for example:

macro= 16  E= -1171.43462197624  delta_E= 2.1746e-09  |g|= 3.90934e-05  1 KF 9 JK
macro= 17  E= -1171.43462197647  delta_E= -2.23281e-10  |g|= 1.75659e-05  1 KF 10 JK
Canonicalize SCF orbitals
macro X = 18  E=-1171.43462197647  |g|= 1.75659e-05  total 44 KF 302 JK
converged SCF energy = -1171.43462197647
rohf_internal: lowest eigs of H = [-0.29657194 -0.06225346  0.00713301]
<class 'pyscf.soscf.newton_ah.SecondOrderROHF'> wavefunction has an internal instability

I wonder if this is because by some approximation in the SOSCF algorithm?

**Comment by jamesETsmith** [comment]
@zh-yichi thanks for sharing. Could you please post a minimal working example so maintainers can try to reproduce your problem?

**Comment by jeanwsr** [comment]
SOSCF does not guarantee the internal stability.
